# motor name. should match an actuator name in a transmission interface
test_joint_motor:
  serial_number: '682129000748' # epos's serial number in hex ('0x' is not required)
  clear_faults: true # clear faults recorded in epos on startup
  rw_ros_units: true # use ros standard units (rad, rad/s, Nm) in topics
                     # or epos standard units (quad count of encoder pulse(qc), rpm, mNm)

  # map from ros_control's controller to epos's operation mode
  operation_mode_map: 
    'velocity_controller': 'profile_velocity'
    'position_controller': 'profile_position'
    'effort_controller': 'current'

  # motor parameters
  motor:
    type: 10 # 1: DC, 10: EC sinus, 11: EC block
    nominal_current: 8.0 # [A]
    max_output_current: 10.0 # [A]
    thermal_time_constant: 3.84 # thermal time constant of motor widing [s]
    number_of_pole_pairs: 2 # EC only

  # power supply parameters. optional
  power_supply:
    name: 'battery' # if not empty, sensor_msgs::BatteryState will be published in 'battery' topic
    location: 'battery_slot' # corresponds BatteryState::location
    technology: 2 # corresponds BatteryState::power_supply_technology
    serial_number: 'unknown' # corresponds BatteryState::serial_number

  # sensor parameters
  sensor:
    type: 1 # 1: INC 3CH, 2: INC 2CH, 4: SSI ABS BIN, 5: SSI ABS GRY
    resolution: 500 # encoder resolution
    inverted_polarity: false

  # safety parameters
  safety:
    max_following_error: 20000 # permitted error [qc] in profile_position mode
    max_profile_velocity: 12000 # max velocity [rpm] in profile_position and profile_velocity modes
    max_acceleration: 15000 # max acceleration [rpm/s] in profile_position nad profile_velocity modes

  # profile position mode configs
  position_profile:
    velocity: 10000 # [rpm]
    acceleration: 8000 # [rpm/s]
    deceleration: 9000 # [rpm/s]

  # profile velocity mode configs
  velocity_profile:
    acceleration: 8000 # [rpm/s]
    deceleration: 9000 # [rpm/s]
